lib/rules/optimal-quantifier-concatenation.ts

```diff
@@ -64,7 +68,11 @@ function getSingleConsumedChar(
     element: Element | Alternative,
     flags: ReadonlyFlags,
 ): SingleConsumedChar {
-    const empty = flags.unicode ? EMPTY_UNICODE : EMPTY_UTF16
+    const empty = flags.unicodeSets
+        ? EMPTY_UNICODE_SETS
+        : flags.unicode
+        ? EMPTY_UNICODE
+        : EMPTY_UTF16
```

Comment:
The maximum character for Unicode and UnicodeSets mode is the same, so `EMPTY_UNICODE.char === EMPTY_UNICODE_SETS.char`. So we don't actually need `EMPTY_UNICODE_SETS` at all.

```suggestion
    const empty = flags.unicode || flags.unicodeSets
        ? EMPTY_UNICODE
        : EMPTY_UTF16
```